src/frontend/Errors.ml

```diff
@@ -1,34 +1,30 @@
 (** Setup of our compiler errors *)
 
 open Core_kernel
-open Middle
 
 (** Our type of syntax error information *)
 type parse_error =
-  | Lexing of string * location
-  | Include of string * location
-  | Parsing of string * location_span
+  | Lexing of string * Middle.location
+  | Include of string * Middle.location
+  | Parsing of string * Middle.location_span
 
 (** Exception for Syntax Errors *)
 exception SyntaxError of parse_error
 
 (** Exception [SemanticError (msg, loc)] indicates a semantic error with message
     [msg], occurring in location [loc]. *)
-exception SemanticError of (string * location_span)
+exception SemanticError of (string * Middle.location_span)
 
 (** Exception [FatalError [msg]] indicates an error that should never happen with message
     [msg]. *)
 exception FatalError of string
 
-(* A semantic error reported by the toplevel *)
-let semantic_error ~loc msg = raise (SemanticError (msg, loc))
-
 (* A fatal error reported by the toplevel *)
 let fatal_error ?(msg = "") _ =
   raise (FatalError ("This should never happen. Please file a bug. " ^ msg))
 
 (** Parse a string into a location *)
-let rec parse_location str =
+let rec parse_location str : Middle.location =
```

Comment:
Why add type signatures?

Reply:
I added when type inference was going a bit nuts; should be ok to remove